app/controllers/admin/cycles_controller.rb

```diff
@@ -72,9 +72,22 @@ def new
 
   def create
```

Comment:
This is becoming troublesome. There are a lot of business logic here on the controller. We should consider moving these pieces to `use_cases`, `repositories`, etc. This will improve readability and mainly tests.

Perhaps add an issue so we can work on this later?